overlord/ifacestate/ifacestate_test.go

```diff
@@ -3078,6 +3078,34 @@ func (s *interfaceManagerSuite) TestSetupProfilesHonorsDevMode(c *C) {
 	c.Check(s.secBackend.SetupCalls[0].Options, Equals, interfaces.ConfinementOptions{DevMode: true})
 }
 
+func (s *interfaceManagerSuite) TestSetupProfilesSetupManyError(c *C) {
+	s.secBackend.SetupCallback = func(snapInfo *snap.Info, opts interfaces.ConfinementOptions, repo *interfaces.Repository) error {
+		return fmt.Errorf("fail")
+	}
+
+	s.MockModel(c, nil)
+
+	// Put the OS snap in place.
+	_ = s.manager(c)
+
+	snapInfo := s.mockSnap(c, sampleSnapYaml)
+
+	// Run the setup-profiles task and let it finish.
+	change := s.addSetupSnapSecurityChange(c, &snapstate.SnapSetup{
+		SideInfo: &snap.SideInfo{
+			RealName: snapInfo.SnapName(),
+			Revision: snapInfo.Revision,
+		},
+	})
+	s.settle(c)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	c.Check(change.Status(), Equals, state.ErrorStatus)
+	c.Check(change.Err(), ErrorMatches, `cannot perform the following tasks:\n-  \(fail\)`)
+}
+
```

Comment:
A test for `if len(snaps) != len(opts) {` would be good since you are doing the check. Presumably it was added because it is conceivable that the case could be hit, so having a test to ensure that check isn't dropped down the line would be good.

Reply:
Test added. It can only be tested directly though, so I exported this helper for tests.